x-pack/legacy/plugins/siem/public/components/navigation/tab_navigation/index.tsx

```diff
@@ -36,71 +35,51 @@ const TabContainer = styled.div`
 
 TabContainer.displayName = 'TabContainer';
 
-interface TabNavigationState {
-  selectedTabId: string;
-}
-
-export class TabNavigation extends React.PureComponent<TabNavigationProps, TabNavigationState> {
-  constructor(props: TabNavigationProps) {
-    super(props);
-    const selectedTabId = this.mapLocationToTab(props.pageName, props.tabName);
-    this.state = { selectedTabId };
-  }
-  public componentWillReceiveProps(nextProps: TabNavigationProps): void {
-    const selectedTabId = this.mapLocationToTab(nextProps.pageName, nextProps.tabName);
-
-    if (this.state.selectedTabId !== selectedTabId) {
-      this.setState(prevState => ({
-        ...prevState,
-        selectedTabId,
-      }));
-    }
-  }
-  public render() {
-    const { display = 'condensed' } = this.props;
-    return (
-      <EuiTabs display={display} size="m">
-        {this.renderTabs()}
-      </EuiTabs>
-    );
-  }
-
-  public mapLocationToTab = (pageName: string, tabName?: HostsTableType): string => {
-    const { navTabs } = this.props;
+export const TabNavigation = React.memo<TabNavigationProps>(props => {
+  const { display = 'condensed', navTabs, pageName, showBorder, tabName } = props;
```

Comment:
nit - to be cleaner
```
React.memo<TabNavigationProps>(({ display = 'condensed', navTabs, pageName, showBorder, tabName }) => {
....
```

Reply:
yes not sure why i did that everywhere else but not here 🤷‍♂

Reply:
nope, we need the `props` value to pass into the `getSearch` function

Reply:
make sense, I did not see it